### ISSUE TITLE
Compatibility issues with DRF

### ISSUE THREAD

**Issue by dgvsecu** [jazzband/django-axes#471]
I am using:

```
djangorestframework==3.9.4
django-axes==5.0.9
Django==2.2.3
```
Problem seems to be here:

https://github.com/jazzband/django-axes/blob/master/axes/helpers.py#L371

```
def get_client_cache_key(request_or_attempt: Union[HttpRequest, Any], credentials: dict = None) -> str:
    """
    Build cache key name from request or AccessAttempt object.
    :param request_or_attempt: HttpRequest or AccessAttempt object
    :param credentials: credentials containing user information
    :return cache_key: Hash key that is usable for Django cache backends
    """

    if isinstance(request_or_attempt, HttpRequest):
        username = get_client_username(request_or_attempt, credentials)
        ip_address = get_client_ip_address(request_or_attempt)
        user_agent = get_client_user_agent(request_or_attempt)
    else:
        username = request_or_attempt.username
        ip_address = request_or_attempt.ip_address
        user_agent = request_or_attempt.user_agent

    filter_kwargs = get_client_parameters(username, ip_address, user_agent)

    cache_key_components = ''.join(value for value in filter_kwargs.values() if value)
    cache_key_digest = md5(cache_key_components.encode()).hexdigest()
    cache_key = f'axes-{cache_key_digest}'
```

Instead of having an `django.http.request.HttpRequest` I have an `rest_framework.request.Request`

If I refactor:

```
from rest_framework.request import Request

...


def get_client_cache_key(request_or_attempt: Union[HttpRequest, Any], credentials: dict = None) -> str:
    """
    Build cache key name from request or AccessAttempt object.

    :param request_or_attempt: HttpRequest or AccessAttempt object
    :param credentials: credentials containing user information
    :return cache_key: Hash key that is usable for Django cache backends
    """

    if isinstance(request_or_attempt, (HttpRequest, Request)):
        username = get_client_username(request_or_attempt, credentials)
        ip_address = get_client_ip_address(request_or_attempt)
        user_agent = get_client_user_agent(request_or_attempt)
    else:
        username = request_or_attempt.username
        ip_address = request_or_attempt.ip_address
        user_agent = request_or_attempt.user_agent

    filter_kwargs = get_client_parameters(username, ip_address, user_agent)

    cache_key_components = ''.join(value for value in filter_kwargs.values() if value)
    cache_key_digest = md5(cache_key_components.encode()).hexdigest()
    cache_key = f'axes-{cache_key_digest}'

    return cache_key
```

It works.

But this refactoring creates a funky dependency to rest_framework. How could this be properly implemented?

Note: the problem was not present with:

```
django-axes==5.0.9
```

**Comment by markddavidoff** [comment]
This is mentioned in the docs:
https://django-axes.readthedocs.io/en/latest/6_integration.html?highlight=DRF#integration-with-django-rest-framework

Does this address your issue?

**Comment by aleksihakli** [comment]
Hey, and thanks for reporting! Seems like we could swap out the type check and

```python
if isinstance(request_or_attempt, AccessAttempt):
    ...
else:
    ...
```

Which should check the types about correctly, and also pass with DRF Request objects.

Would you be interested in opening a PR and checking if that would fix the issue?

We can't depend on DRF there so this has to be implemented without it :)

**Comment by aleksihakli** [comment]
Fix has been released in version 5.0.11.

**Comment by dgvsecu** [comment]
> This is mentioned in the docs:
> https://django-axes.readthedocs.io/en/latest/6_integration.html?highlight=DRF#integration-with-django-rest-framework
> 
> Does this address your issue?

Not, it is unrelated.

I have anyway a recent DRF, as stated in my explanation.

**Comment by dgvsecu** [comment]
> Fix has been released in version 5.0.11.

Wow, thanks, that was quick! It has even hit pypi! :heart_eyes:
